packages/dds/tree/api-report/tree.api.md

```diff
@@ -4,6 +4,8 @@
 
 ```ts
 
+import { FieldKind as FieldKind_2 } from './schemaTypes.js';
+import { FieldSchema as FieldSchema_2 } from './schemaTypes.js';
 import { FluidObject } from '@fluidframework/core-interfaces';
```

Comment:
These are not valid which means the API extractor generated .d.ts files are probably incorrect.

Reply:
Here is an example log when I had such gunk: [https://dev.azure.com/fluidframework/public/_build/results?buildId=228628&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=bfe27d5f-9219-57c0-0d67-1b2946a32921](https://dev.azure.com/fluidframework/public/_build/results?buildId=228628&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=bfe27d5f-9219-57c0-0d67-1b2946a32921)

Reply:
It's worrisome to me that in the example you linked, the tools involved were only tsc and api-extractor. Does that imply that api-extractor itself doesn't like the Tree types? Maybe we should try updating api-extractor? Looks like the recent minor release added support for TypeScript 5.3 - https://github.com/microsoft/rushstack/blob/main/apps/api-extractor/CHANGELOG.md#7390

@Josmithr FYI.

Reply:
I was guessing it isn't a Tree type thing but something with moduleResolution support.

I did a test with updated api-extractor for my branch and results were not great.
First I tried exactly as I had things and then with tree typescript set to 5.3.3 which the changelog said is supported:
```log
@fluidframework/tree: error during command 'api-extractor run --local --config ./api-extractor-lint.json' (exit code 1)
@fluidframework/tree: api-extractor 7.39.1  - https://api-extractor.com/
@fluidframework/tree: 
@fluidframework/tree: 
@fluidframework/tree: 
@fluidframework/tree: ERROR: Error parsing /home/jasonha/ff/FluidFramework/packages/dds/tree/api-extractor-lint.json:
@fluidframework/tree: The "bundledPackages" list contains an invalid package name: "@fluidframework/.*"
[ 8/12] x @fluidframework/tree: api-extractor run --local --config ./api-extractor-lint.json - 0.327s
@fluidframework/tree: error during command 'api-extractor run --config ./api-extractor-cjs.json' (exit code 1)
@fluidframework/tree: api-extractor 7.39.1  - https://api-extractor.com/
@fluidframework/tree: 
@fluidframework/tree: Analysis will use the bundled TypeScript version 5.1.6
@fluidframework/tree: *** The target project appears to use TypeScript 5.3.3 which is newer than the bundled compiler engine; consider upgrading API Extractor.
@fluidframework/tree: 
@fluidframework/tree: 
@fluidframework/tree: ERROR: program.getResolvedModule is not a function
[ 9/12] x @fluidframework/tree: api-extractor run --config ./api-extractor-cjs.json - 2.742s
@fluidframework/tree: error during command 'api-extractor run --local --config ./api-extractor-esm.json' (exit code 1)
@fluidframework/tree: api-extractor 7.39.1  - https://api-extractor.com/
@fluidframework/tree: 
@fluidframework/tree: Analysis will use the bundled TypeScript version 5.1.6
@fluidframework/tree: *** The target project appears to use TypeScript 5.3.3 which is newer than the bundled compiler engine; consider upgrading API Extractor.
@fluidframework/tree: 
@fluidframework/tree: 
@fluidframework/tree: ERROR: program.getResolvedModule is not a function
[10/12] x @fluidframework/tree: api-extractor run --local --config ./api-extractor-esm.json - 2.726s
```

Reply:
These errors might be related to the manual api-extractor patch that @Josmithr has been managing. Presumably that patch will need to be regenerated.

Reply:
Good call out. The patch is listed as for a specific version but might not be. I then also found that we were forcing 5.1.6 into api-extractor per pnpm override. When I removed those two things the main runs worked. The lint case still complains about malformed bundledPackages.
No change on the results for the different module resolutions and bad results.
I feel like I've seen such things in other packages, but maybe it was always when playing with module resolution. I think there was some checked in at some point and then something corrected it later.

Reply:
Ha, ha. The patch is specifically about making `bundledPackages` support regex.